### ISSUE TITLE
Fine-tune text detector

### ISSUE THREAD

**Issue by wesamalnabki** [mindee/doctr#1564]
### Bug description

I'm trying to fine-tune the text detector model as described in this [example](https://www.mindee.com/blog/vin-extraction-with-doctr) but facing an error

### Code snippet to reproduce the bug

```
python references/detection/train_pytorch.py \
"/opt/XXX/OCR_Dataset/detection_train_tmp" \
"/opt/XXX/OCR_Dataset/detection_val_tmp" \
db_resnet50 \
--name 23042023-1650-det \
--epochs 5 \
--pretrained --wb \
--batch_size 2 \
--pretrained
```

### Error traceback

![image](https://github.com/mindee/doctr/assets/6717881/40fc3f92-abd8-4d31-8f5d-eb07a6a78d30)
 

### Environment

Collecting environment information...

DocTR version: v0.8.1
TensorFlow version: N/A
PyTorch version: 2.2.2+cu121 (torchvision 0.17.2+cu121)
OpenCV version: 4.9.0
OS: Ubuntu 20.04.6 LTS
Python version: 3.10.14
Is CUDA available (TensorFlow): N/A
Is CUDA available (PyTorch): Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Tesla V100-PCIE-16GB
Nvidia driver version: 535.171.04
cuDNN version: Could not collect

### Deep Learning backend

>>> print(f"is_tf_available: {is_tf_available()}")
is_tf_available: False
>>>
>>> print(f"is_torch_available: {is_torch_available()}")
is_torch_available: True

**Comment by felixdittrich92** [comment]
Hi @wesamalnabki :wave:,

Does your dataset contain maybe empty annotations or samples which contains only super small polygon annotations ?
Because this will only happen if we can't build the thresh mask which is required to compute the l1_loss

**Comment by wesamalnabki** [comment]
Thanks for your reply @felixdittrich92 
Actually, there are no empty annotations, and each bbox has the four coordinates of the bounding box.

Here is an example of the dataset I'm using. This colab  download an image of the dataset and its bboxs, and visualize them. 

https://colab.research.google.com/drive/16Aabf0kJxxYov-XRg3VO3joeUjLH2-7A#scrollTo=q2qPZdwayfnZ

I hosted a subset of the dataset (~10 images) here
https://drive.google.com/drive/folders/1P1-Or7eCa8lFFvx41EH5uvdFarjfvUsB?usp=drive_link

**Comment by felixdittrich92** [comment]
Hi @wesamalnabki Thanks for sharing i will have a look :+1: 

Looks like https://huggingface.co/datasets/pixparse/pdfa-eng-wds right ? ^^

**Comment by wesamalnabki** [comment]
yes, exactly, its a massive OCR dataset. 
Actually, I want to fine-tune to enhance the vertical text detection.

**Comment by felixdittrich92** [comment]
short update

label annotations are wrong:

![Screenshot from 2024-04-24 13-42-09](https://github.com/mindee/doctr/assets/23098070/61374a26-aefc-41a8-a1d5-3ba7f2f5da45)

**Comment by wesamalnabki** [comment]
regarding the first points, I provided the coordinates like doctr format, I adjusted the pdf format, but the error was because the bbox was normalized.  I removed the bbox normalization, and the training is working now, let's see.

**Comment by felixdittrich92** [comment]
:+1: 

We have also planned to experiment with the dataset in near future so if you could share some insights that would be super helpful :)

CC @odulcy-mindee

**Comment by wesamalnabki** [comment]
Finally, the training on a small subset of the dataset worked well. The problem was the normalization of the bboxs.
Once I have trained on the full dataset, I will share with you the results ;)

**Comment by wesamalnabki** [comment]
Sure, here is the script

https://gist.github.com/wesamalnabki/06fdd17b06adca79a0cd459f0b875298

**Comment by felixdittrich92** [comment]
https://github.com/mindee/doctr/discussions/1565

**Comment by felixdittrich92** [comment]
@wesamalnabki i quickly cheked your script and you have still an minor but significant issue in your script.
change to:

```
def adjust_cord(bbox, img_w, img_h):
    x, y, w, h = bbox
    x1, y1 = x, y
    x2, y2 = x + w, y
    x3, y3 = x + w, y + h
    x4, y4 = x, y + h
    return [
        [int(x1 * img_w), int(y1 * img_h)],
        [int(x2 * img_w), int(y2 * img_h)],
        [int(x3 * img_w), int(y3 * img_h)],
        [int(x4 * img_w), int(y4 * img_h)]
    ]
```
with `--show-samples` now it's correct:

![Screenshot from 2024-04-24 16-08-54](https://github.com/mindee/doctr/assets/23098070/35b119d3-9938-4ac0-bd38-bb869d53239a)
